### PR TITLE
Update capz crds to v1alpha4

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/18235

As part of the migration to upstream CAPI/CAPZ controllers, we need to migrate our CRD's to the latest versions. This `v1alpha4` version is using a conversion webhook upstream, and it's marked as the stored version in etcd. For our migration, we want to apply these CRDs first, but removing the webhook and keeping `v1alpha3` as the one stored in etcd.

Once we do this, we can [deploy the webhook](https://github.com/giantswarm/giantswarm/issues/18234).

Then we can update the CRDs on this repository again, adding back the conversion webhook.

Finally we only have to change the stored version from `v1alpha3` to `v1alpha4`, and we'd be using exactly the same as upstream.

## Checklist

- [ ] Consider SIG UX feedback.
- [X] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/crds-azure/templates/upstream.yaml [hunk 1]
```diff
@@ -137,10 +262,35 @@ spec:
...
+    - jsonPath: .spec.location
+      name: Location
+      priority: 1
+      type: string
+    - description: Control Plane Endpoint
+      jsonPath: .spec.controlPlaneEndpoint.host
+      name: Endpoint
+      priority: 1
+      type: string
+    name: v1alpha3
```
Comment (fiunchinho) [inline helm/crds-azure/templates/upstream.yaml:290]: v1alpha2 is removed and v1alpha3 is added.

File: helm/crds-azure/templates/upstream.yaml [hunk 2]
```diff
@@ -253,203 +504,65 @@ spec:
...
+                description: 'FailureDomains specifies the list of unique failure domains for the location/region of the cluster. A FailureDomain maps to Availability Zone with an Azure Region (if the region support them). An Availability Zone is a separate data center within a region and they can be used to ensure the cluster is more resilient to failure. See: https://docs.microsoft.com/en-us/azure/availability-zones/az-overview This list will be used by Cluster API to try and spread the machines across the failure domains.'
                 type: object
               ready:
                 description: Ready is true when the provider resource is ready.
                 type: boolean
             type: object
         type: object
     served: true
-    storage: false
+    storage: true
```
Comment (fiunchinho) [inline helm/crds-azure/templates/upstream.yaml:565]: v1alpha3 is the one stored

File: helm/crds-azure/templates/upstream.yaml [hunk 3]
```diff
@@ -744,48 +1184,795 @@ spec:
...
+                description: 'FailureDomains specifies the list of unique failure domains for the location/region of the cluster. A FailureDomain maps to Availability Zone with an Azure Region (if the region support them). An Availability Zone is a separate data center within a region and they can be used to ensure the cluster is more resilient to failure. See: https://docs.microsoft.com/en-us/azure/availability-zones/az-overview This list will be used by Cluster API to try and spread the machines across the failure domains.'
+                type: object
+              ready:
+                description: Ready is true when the provider resource is ready.
+                type: boolean
+            type: object
+        type: object
+    served: true
+    storage: false
```
Comment (fiunchinho) [inline helm/crds-azure/templates/upstream.yaml:1195]: v1alpha4 is not stored

File: helm/crds-azure/templates/upstream.yaml [hunk 4]
```diff
@@ -744,48 +1184,795 @@ spec:
...
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    cert-manager.io/inject-ca-from: capz-system/capz-serving-cert
+    controller-gen.kubebuilder.io/version: v0.6.1
+  creationTimestamp: null
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-azure
+    cluster.x-k8s.io/v1alpha4: v1alpha4
+  name: azuremachinepools.infrastructure.cluster.x-k8s.io
```
Comment (fiunchinho) [inline helm/crds-azure/templates/upstream.yaml:1573]: This is not the same as our previous `AzureMachinePool`. This is using a different api group.

File: helm/crds-azure/templates/upstream.yaml [hunk 5]
```diff
@@ -744,48 +1184,782 @@ spec:
...
+                type: boolean
+              version:
+                description: Version defines the Kubernetes version for the VM Instance
+                type: string
+            required:
+            - latestModelApplied
+            type: object
+        type: object
+    served: true
+    storage: true
```
Comment (fiunchinho) [inline helm/crds-azure/templates/upstream.yaml:1552]: This one is stored as v1alpha4 because there is no v1alpha3, is a new crd.

**Comment by tfussell** [comment]
The verification failure should be fixed now if you merge master into this branch and fix the merge conflicts.

**Comment by tuommaki** [comment]
> The verification failure should be fixed now if you merge master into this branch and fix the merge conflicts.

I did this, but now the build is broken locally:

```
$ make
find: warning: you have specified the global option -depth after the argument -name, but global options are not positional, i.e., -depth affects tests specified before it as well as those specified after it.  Please specify global options before other arguments.
find: paths must precede expression: `3'
====> apiextensions-v3.32.0-5351d569231075740d7919b5b4708c1a3ae3a2cd-linux-amd64
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-w -linkmode 'auto' -extldflags '-static' -X 'github.com/giantswarm/apiextensions/v3/pkg/project.buildTimestamp=2021-08-16T08:30:29Z' -X 'github.com/giantswarm/apiextensions/v3/pkg/project.gitSHA=5351d569231075740d7919b5b4708c1a3ae3a2cd'" -o apiextensions-v3.32.0-5351d569231075740d7919b5b4708c1a3ae3a2cd-linux-amd64 .
no Go files in /home/tuommaki/go/src/github.com/giantswarm/apiextensions
make: *** [Makefile.gen.go.mk:58: apiextensions-v3.32.0-5351d569231075740d7919b5b4708c1a3ae3a2cd-linux-amd64] Error 1
````

**Comment by tfussell** [comment]
> > The verification failure should be fixed now if you merge master into this branch and fix the merge conflicts.
> 
> I did this, but now the build is broken locally:

Please use `make generate` instead of just `make`. The two Makefiles are interfering on the default rule.

**Comment by fiunchinho** [comment]
@tfussell I'm still getting ordering errors during CI after running `make generate` locally on my machine https://github.com/giantswarm/apiextensions/pull/769/checks?check_run_id=3361264708

**Comment by fiunchinho** [comment]
Superseeded by https://github.com/giantswarm/apiextensions/pull/855

**Pull request closed without merging**